PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -3253,17 +3272,99 @@ struct HfTrackIndexSkimCreatorLfCascades {
-          if (trackPion1.globalIndex() == trackXiDauCharged.globalIndex() || trackPion1.globalIndex() == trackV0DauPos.globalIndex() || trackPion1.globalIndex() == trackV0DauNeg.globalIndex() || trackPion1.globalIndex() == casc.globalIndex()) {
+          if (trackPion1.globalIndex() == trackXiDauCharged.globalIndex() || trackPion1.globalIndex() == trackV0PosDau.globalIndex() || trackPion1.globalIndex() == trackV0NegDau.globalIndex()) {
             continue;
           }
 
           // primary pion track to be processed with DCAFitter
           auto trackParVarPion1 = getTrackParCov(trackPion1);
 
+          // find charm baryon decay using xi PID hypothesis
+          int nVtxFrom2ProngFitterXiHyp = df2.process(trackCascXi, trackParVarPion1);
+          if (nVtxFrom2ProngFitterXiHyp > 0) {
+
+            statusMassXiPiChannel = true;
+
+            df2.propagateTracksToVertex();
+
+            std::array<float, 3> pVecXi = {0.};
+            std::array<float, 3> pVecPion1XiHyp = {0.};
+            df2.getTrack(0).getPxPyPzGlo(pVecXi);
+            df2.getTrack(1).getPxPyPzGlo(pVecPion1XiHyp);
+
+            auto secondaryVertex2XiHyp = df2.getPCACandidate();
+
+            if (std::abs(casc.mXi() - massXi) < cascadeMassWindow) {
+              SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi);
+            }
+
+            std::array<std::array<float, 3>, 2> arrMomToXi = {pVecXi, pVecPion1XiHyp};
+            auto mass2ProngXiHyp = RecoDecay::m(arrMomToXi, arrMass2Prong[hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi]);
+
+            if (mass2ProngXiHyp < massXiPiMin || mass2ProngXiHyp > massXiPiMax) {
+              statusMassXiPiChannel = false;
+            }
+
+            // fill histograms
+            if (fillHistograms && statusMassXiPiChannel) {
+              registry.fill(HIST("hVtx2ProngXXiHyp"), secondaryVertex2XiHyp[0]);
+              registry.fill(HIST("hVtx2ProngYXiHyp"), secondaryVertex2XiHyp[1]);
+              registry.fill(HIST("hVtx2ProngZXiHyp"), secondaryVertex2XiHyp[2]);
+              registry.fill(HIST("hMassXicZeroOmegacZeroToXiPi"), mass2ProngXiHyp);
+            }
+          }
+
+          // find charm baryon decay using omega PID hypothesis
+          int nVtxFrom2ProngFitterOmegaHyp = df2.process(trackCascOmega, trackParVarPion1);
+          if (nVtxFrom2ProngFitterOmegaHyp > 0) {
+
+            statusMassOmegaPiChannel = true;
+
+            df2.propagateTracksToVertex();
+
+            std::array<float, 3> pVecOmega = {0.};
+            std::array<float, 3> pVecPion1OmegaHyp = {0.};
+            df2.getTrack(0).getPxPyPzGlo(pVecOmega);
+            df2.getTrack(1).getPxPyPzGlo(pVecPion1OmegaHyp);
+
+            auto secondaryVertex2OmegaHyp = df2.getPCACandidate();
+
+            if (std::abs(casc.mOmega() - massOmega) < cascadeMassWindow) {
+              SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaPi);
```

Comment:
Same here, this should be only set after you test (mass2ProngOmegaHyp < massOmegaPiMin || mass2ProngOmegaHyp > massOmegaPiMax) (line 3304 of current draft)